src/commcare_cloud/fab/utils.py

```diff
@@ -273,3 +280,76 @@ def generate_bower_command(command, production=True, config=None):
 def bower_command(command, production=True, config=None):
     cmd = generate_bower_command(command, production, config)
     sudo(cmd)
+
+def warn_of_migrations(last_deploy_sha, current_deploy_sha):
+    pr_numbers = _get_pr_numbers(last_deploy_sha, current_deploy_sha)
+    pool = Pool(5)
+    pr_infos = [_f for _f in pool.map(_get_pr_info, pr_numbers) if _f]
+
+    print("List of PRs since last deploy:")
+    _print_prs_formatted(pr_infos)
+
+    prs_by_label = _get_prs_by_label(pr_infos)
+    if prs_by_label:
+        print(red('You are about to deploy the following PR(s), which will trigger a reindex or migration. Click the URL for additional context.'))
+        _print_prs_formatted(prs_by_label['reindex/migration'])
+
+def _get_pr_numbers(last_deploy_sha, current_deploy_sha):
+    repo = _get_github().get_organization('dimagi').get_repo('commcare-hq')
+    comparison = repo.compare(last_deploy_sha,current_deploy_sha)
+
+    return [
+        int(re.search(r'Merge pull request #(\d+)', repo_commit.commit.message).group(1))
+        for repo_commit in comparison.commits
+        if repo_commit.commit.message.startswith('Merge pull request')
+    ]
+
+def _get_pr_info(pr_number):
+    url = 'https://api.github.com/repos/dimagi/commcare-hq/pulls/{}'.format(pr_number)
+    json_response = requests.get(url).json()
+    if not json_response.get('number'):
+        return None
+    assert pr_number == json_response['number'], (pr_number, json_response['number'])
+
+    labels = _get_pr_labels(pr_number)
+
+    return {
+        'title': json_response['title'],
+        'url': json_response['html_url'],
+        'labels': labels,
+    }
+
+def _get_pr_labels(pr_number):
+    url = 'https://api.github.com/repos/dimagi/commcare-hq/issues/{}'.format(pr_number)
+    json_response = requests.get(url).json()
+    if not json_response.get('number'):
+        return []
+    assert pr_number == json_response['number'], (pr_number, json_response['number'])
+
+    return [
+        {'name': label['name'], 'color': label['color']}
+        for label in json_response['labels']
+    ]
+
+def _get_prs_by_label(pr_infos):
+    prs_by_label = defaultdict(list)
+    for pr in pr_infos:
+        for label in pr['labels']:
+            if label['name'] in LABELS_TO_EXPAND:
+                prs_by_label[label['name']].append(pr)
+    return dict(prs_by_label)
+
+def _print_prs_formatted(pr_list):
+    i = 1
+    for pr in pr_list:
+        print("{0}. ".format(i), end="")
+        print ("{title} {url} | ".format(**pr), end="")
+        j = 0
+        num_labels = len(pr['labels'])
+        for label in pr['labels']:
+            j += 1
+            if (j == num_labels):
+                print("{name}".format(**label))
+            else:
+                print ("{name} ".format(**label), end="")
```

Comment:
an easier way to do this is:
```python
print(" ".join(label['name'] for label in pr['labels']))
```

Reply:
Nice, the way I was doing it before seemed rather non-pythonic. Thanks for the tip!